### ISSUE TITLE
Voltage indicating incorrectly, "module not support", aggregated power feature request

### ISSUE THREAD

**Issue by teshreve** [jamesbarnett91/tplink-energy-monitor#7]
Thank you for this project.

Not sure what is protocol here, so just barfing all my feedback into a single post.
Firstly, I have an inordinate about of TPLink devices, including LB120, LB130, 2x HS100, 10x HS110, and an HS200. I searched for and discovered this project primarily to be able to easily view the aggregate power usage of all (or some) my HS110 plugs.

My feedback
1) All my HS110 plugs are connected to 240V power. This has been verified by a multimeter. As well, I did the wiring, breaker installs, etc myself ;) But the application is reporting ~170v. The HS100 plugs, connected to 120v are reporting around 86v. Which is interesting as the Android app doesn't present voltage at all. Perhaps it is not show in the Android app because the voltage is broken in the devices. And perhaps this problem is totally unrelated to your project.  Because watts = volts * amps, this is all easy to validate that something is wrong. For example, one of my HS110 shows: 760w, 3.15a, 172v. The math doesn't quite work out here. And indeed, 760/3.15=241. So I trust the amps and watts, but the volts is wack.

2) Likely a result of the app discovering the bulbs (LB120, LB130), this error scrolls by endlessly: **(node:11096) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 4103): ResponseError: err_code not zero response: {"err_code":-1,"err_msg":"module not support"}**

3)  I am unable to contribute with code (although happy to contribute with money or beer), but what I want more than anything is a quick way to see how much power all the HS110s are using. Bonus: Select the HS110s I want to aggregate for a sub-total power figure.

4) I place all my IoT devices on their own SSID, their own IP network, and in their own security zone on my firewall. Thus only by connecting my PC to this network can I discover the devices. Which is counter to the entire reason for having an IoT network: Security. It would seem better to auth to the web API?

Thanks again!

**Comment by jamesbarnett91** [comment]
Hey, thanks for the detailed feedback!
Regarding your issues:

1. I've just noticed this as well after updating the firmware version of my HS110 to v1.2.5. I was previously running version 1.0 and and the plug seemed to be reporting the voltage as its peak to peak value rather than RMS (e.g 310v rather than 240v), so I added logic to calculate the RMS value from the reported voltage. Now it seems like later firmware versions return the value correctly in RMS, which is why the dashboard now shows ~170v (240 / sqrt(2)). I'll remove this extra calculation if the plugs firmware is > 1.0, which should then show the correct voltage in all cases.

2. I'm only scanning for plug type devices, so my guess is that error message is coming from your HS100. Someone else reported this as well so I'll try and add some logic to filter out HS100s and not request their energy data.

3. Agree that this would be a nice feature to have, and I'd find it useful myself. Im adding some logging capabilities currently, but I'll look at adding a screen showing aggregated data from all plugs as the next feature. It shouldn't be too hard to add since all the functionality is there.

4. I personally don't use any of the cloud based functionality of the plugs, and only access them locally on my network, so I don't have much interest in using the web API. 
It would be a non-trivial change to add a new implementation of the data-fetcher that used the web API, and there would need to be some kind of set-up process where the user could enter their API key. The app can end up making a reasonably high volume of requests as well, so I don't know if there would be any kind of rate limiting issues.

If anyone else is willing to try and implement it then I'd be happy to merge a PR, but it's not something I really have the time or motivation to do myself.

I'll close this issue once the bugs in 1 and 2 are fixed, and raise a separate issue for the aggregated usage page.

Thanks for using the project!
